### PR TITLE
[spirv] Emit counter variables for alias structured/byte buffers

### PR DESCRIPTION
**Pull request by antiagainst**
Also handled updating counter variable aliases in assignment and
function calls.

### CODE REVIEW

**Comment by antiagainst** [comment]
This is the second PR regarding legalizing structured/byte buffer aliasing.

Only the last one needs review; all previous commits are in #918.

**Comment by marcelolr** [comment]
The variability on AppVeyor builds seems pretty crazy. https://ci.appveyor.com/project/marcelolr/directxshadercompiler/build/1.0.1590 timed out, but https://ci.appveyor.com/project/marcelolr/directxshadercompiler/build/1.0.1583 completed in 41 minutes!

**Comment by antiagainst** [comment]
Yep. It truly varies a lot.

It appears Appveyor are using two clouds: https://www.appveyor.com/docs/build-environment/#build-clouds. Maybe that's one of the reason?

File: tools/clang/lib/SPIRV/DeclResultIdMapper.h [hunk 1]
```diff
@@ -180,6 +212,12 @@ class DeclResultIdMapper {
   /// returns its <result-id>.
   uint32_t createFnParam(const ParmVarDecl *param);
 
+  /// \brief Creates the counter variable associated with the givne param.
```
[the remark this answers is not in the document]
Comment (antiagainst) [reply tools/clang/lib/SPIRV/DeclResultIdMapper.h]: Done

File: tools/clang/lib/SPIRV/SPIRVEmitter.h [hunk 2]
```diff
@@ -648,7 +648,15 @@ class SPIRVEmitter : public ASTConsumer {
 
   /// \brief Increments or decrements the counter for RW/Append/Consume
   /// structured buffer.
```
[the remark this answers is not in the document]
Comment (antiagainst) [reply tools/clang/lib/SPIRV/SPIRVEmitter.h]: Done

**Comment by antiagainst** [comment]
Waiting on KhronosGroup/SPIRV-Tools#1096 to submit

**Pull request merged**